### PR TITLE
Zppy pcmdi new

### LINKED ISSUE #141: Add task for PMP
Add task for PCMDI Metrics Package (PMP): https://climatemodeling.science.energy.gov/e3sm/pmp-0181-0200

### PR DESCRIPTION
**Pull request by forsyth2**
## Summary

This is an alternative to #656 that wants to merge to `main` rather than `zppy_pcmdi` (which is #647's branch)

Objectives:
- Objective 1
- Objective 2
- ...
- Objective n

Issue resolution:
- Closes #141

Select one: This pull request is...
- [ ] a bug fix: increment the patch version
- [ ] a small improvement: increment the minor version
- [x] a new feature: increment the minor version
- [ ] an incompatible (non-backwards compatible) API change: increment the major version

## Big Change

- [ ] To merge, I will use "Create a merge commit". That is, this change is large enough to require multiple units of work (i.e., it should be multiple commits).

### 1. Does this do what we want it to do?

Required:
- [ ] Product Management: I have confirmed with the stakeholders that the objectives above are correct and complete.
- [ ] Testing: I have added or modified at least one "min-case" configuration file to test this change. Every objective above is represented in at least one cfg.
- [ ] Testing: I have considered likely and/or severe edge cases and have included them in testing.

If applicable:
- [ ] Testing: this pull request introduces an important feature or bug fix that we _must_ test often. I have updated the weekly-test configuration files, not just a "min-case" one.
- [ ] Testing: this pull request adds at least one new possible parameter to the cfg. I have tested using this parameter with and without any other parameter that may interact with it.

### 2. Are the implementation details accurate & efficient?

Required:
- [ ] Logic: I have visually inspected the entire pull request myself.
- [ ] Logic: I have left GitHub comments highlighting important pieces of code logic. I have had these code blocks reviewed by at least one other team member.

If applicable:
- [ ] Dependencies: This pull request introduces a new dependency. I have discussed this requirement with at least one other team member. The dependency is noted in `zppy/conda`, not just an `import` statement.

### 3. Is this well documented?

Required:
- [ ] Documentation: by looking at the docs, a new user could easily understand the functionality introduced by this pull request.

### 4. Is this code clean?

Required:
- [ ] Readability: The code is as simple as possible and well-commented, such that a new team member could understand what's happening.
- [ ] Pre-commit checks: All the pre-commits checks have passed.

If applicable:
- [ ] Software architecture: I have discussed relevant trade-offs in design decisions with at least one other team member. It is unlikely that this pull request will increase tech debt.

### CODE REVIEW

File: conda/chrisalis.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: zppy_dev
```
Comment (forsyth2) [inline conda/chrisalis.yml:1]: This file can probably be removed. Is it used anywhere?

File: zppy/defaults/default.ini [hunk 2]
```diff
@@ -107,9 +107,11 @@ input_component = string(default="")
 [ts]
 area_nm = string(default="area")
 cmip_metadata = string(default="inclusions/e3sm_to_cmip/default_metadata.json")
+cmip_plevdata = string(default="inclusions/e3sm_to_cmip/vrt_remap_plev19.nc")
```
Comment (forsyth2) [inline zppy/defaults/default.ini:110]: This code needs to be rebased off the latest `main`. `cmip_metadata` isn't handled under `ts` anymore, but rather under the new `e3sm_to_cmip` task.

File: zppy/defaults/default.ini [hunk 3]
```diff
@@ -119,18 +121,243 @@ input_component = string(default="")
...
 
 [tc_analysis]
 # NOTE: always overrides value in [default]
 input_files = string(default="eam.h2")
 # The scratch directory
 scratch = string(default="")
 
+[pcmdi_diags]
+backend = string(default="mpl")
+cfg = string(default="")
```
Comment (forsyth2) [inline zppy/defaults/default.ini:140]: Not used; should make sure every parameter is needed

File: zppy/templates/coupled_global.py [hunk 4]
```diff
@@ -0,0 +1,872 @@
+# Script to plot some global atmosphere and ocean time series
```
Comment (forsyth2) [inline zppy/templates/coupled_global.py:1]: This addition undoes the refactor; I also don't think it's used.

File: zppy/templates/inclusions/e3sm_to_cmip/default_metadata.json
Comment (forsyth2) [inline zppy/templates/inclusions/e3sm_to_cmip/default_metadata.json:1]: Remove change

File: zppy/templates/inclusions/e3sm_to_cmip/vrt_remap_plev19.nc
Comment (forsyth2) [inline zppy/templates/inclusions/e3sm_to_cmip/vrt_remap_plev19.nc:1]: Remove change

File: zppy/templates/global_time_series.bash
Comment (forsyth2) [inline zppy/templates/global_time_series.bash:1]: Remove change

File: zppy/templates/pcmdi_diags.bash [hunk 5]
```diff
@@ -0,0 +1,1308 @@
...
+{% raw %}
+ENDTIME=$(date +%s)
+ELAPSEDTIME=$(($ENDTIME - $STARTTIME))
+{% endraw %}
+echo ==============================================
+echo "Elapsed time: $ELAPSEDTIME seconds"
+echo ==============================================
+rm -f {{ prefix }}.status
+echo 'OK' > {{ prefix }}.status
+exit 0
```
Comment (forsyth2) [inline zppy/templates/pcmdi_diags.bash:1308]: We should move some of this into `zppy-interfaces`. We shouldn't have a 1300-line bash file in `zppy`; these bash files should be pretty minimal.

File: zppy/templates/ts.bash [hunk 6]
```diff
@@ -147,19 +147,39 @@ EOF
   cp -s $dest/*_{{ '%04d' % (yr_start) }}??_{{ '%04d' % (yr_end) }}??.nc $input_dir
   dest_cmip={{ output }}/post/{{ component }}/{{ grid }}/cmip_ts/{{ frequency }}
   mkdir -p ${dest_cmip}
+
   {{ e3sm_to_cmip_environment_commands }}
+
```
Comment (forsyth2) [inline zppy/templates/ts.bash:152]: Yes, this needs to be rebased off the refactored code to use the e3sm_to_cmip task.

**Review by forsyth2 — COMMENTED** [review]
@zhangshixuan1987 I made this PR to compare against `main` (see note at https://github.com/E3SM-Project/zppy/pull/656#issuecomment-2902007450), since I'm assuming that's what was actually intended.

I took a very high level look, but we're going to have to get this rebased off the latest `main` before I can really understand what the additions are.

I will try to do what I can in a code clean up commit.

**Comment by forsyth2** [comment]
I actually see that https://github.com/E3SM-Project/zppy/pull/718/commits/e7dda2167c8fc0a2c0145c978f99d16c7b326bd8 is the first commit out of 24 here. That commit is the _only_ one on https://github.com/E3SM-Project/zppy/pull/647/commits, so yes I'm fairly certain the intention was for these commits to go into `main` instead.

**Comment by forsyth2** [comment]
Replaced by #719.
